src/content_sao.cpp

```diff
@@ -1273,6 +1273,38 @@ std::string PlayerSAO::getWieldList() const
 	return "main";
 }
 
+ItemStack PlayerSAO::getWieldedItem() const
+{
+	const Inventory *inv = getInventory();
+	ItemStack ret;
+	const InventoryList *mlist = inv->getList(getWieldList());
+	if (mlist && (getWieldIndex() < (s32)mlist->getSize()))
+		ret = mlist->getItem(getWieldIndex());
+	if (ret.name.empty()) {
+		if (const InventoryList *hlist = inv->getList("hand"))
```

Comment:
I personally do not like assignments in if statements, it's very easy to misread